internal/conf/computed.go

```diff
@@ -434,6 +437,62 @@ func ProductResearchPageEnabled() bool {
 	return true
 }
 
+type lastParsedURLValue struct {
+	url       *url.URL
+	confValue string
+}
+
+var (
+	lastParsedURL *atomic.Pointer[lastParsedURLValue] = atomic.NewPointer(&lastParsedURLValue{
+		url: &url.URL{
+			Scheme: "http",
+			Host:   "example.com",
+		},
+		confValue: "",
+	})
+)
+
+// ExternalURLParsed returns a parsed version of conf.ExternalURL().
+// This function is thread-safe and returns a copy of the cached parsed version
+// of the URL, so it is also safe to mutate.
+func ExternalURLParsed() *url.URL {
+	// Note that we do NOT use a mutex here, and instead let callers parse the URL
+	// simultaneously during the short period where the URL was changed but the new
+	// parsed value has not yet been cached.
+	// This eliminates the need to acquire a mutex entirely, and avoids potential costly
+	// locking if many requests are served concurrently.
+	urlString := ExternalURL()
+	lastParsed := lastParsedURL.Load()
+	if lastParsed.confValue != urlString {
+		parsed, err := url.Parse(urlString)
+		if err != nil {
+			log.Scoped("conf.ExternalURL").Error("failed to parse external URL", log.Error(err), log.String("externalURL", urlString))
+			return cloneURL(lastParsed.url)
+		}
+		lastParsed = &lastParsedURLValue{
+			url:       parsed,
+			confValue: urlString,
+		}
+		lastParsedURL.Store(lastParsed)
```

Comment:
nit: this looks correct to me, but can we just inline this so I don't have to convince myself we're not writing into the `lastParsed` pointer?

```suggestion
		lastParsedURL.store(&lastParsedURLValue{
			url:       parsed,
			confValue: urlString,
		})		
```

Reply:
yes, just need to also return the new parsed value then otherwise it'll return the stale data below 😬

Reply:
Aah, gotcha. Missed why that exists. Ignore me then